django/test/client.py

```diff
@@ -459,18 +468,21 @@ def _get_path(self, parsed):
         # Refs comment in `get_bytes_from_wsgi()`.
         return path.decode("iso-8859-1")
 
-    def get(self, path, data=None, secure=False, *, headers=None, **extra):
+    def get(
+        self, path, data=None, secure=False, *, headers=None, query_params=None, **extra
+    ):
         """Construct a GET request."""
-        data = {} if data is None else data
+        if query_params and data:
+            raise ValueError("You cannot specify query_params and data together.")
```

Comment:
Was this tested?

Reply:
Ok, you caught me. I added tests for this. Do you think we also need tests to make sure you _can_ use both for post/put/patch, or is that unnecessary?

Reply:
The exceptional behaviour seems worth testing. 

A POST with `data` to a URL with `query_params` might be interesting for documentation/regression purposes. (But it's not something I'd insist on: what's your thought?)

Reply:
Considering that's a primary use case of the PR, seems like a good idea. Added tests around this and some others I missed.